app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java

```diff
@@ -0,0 +1,98 @@
+import ch.epfl.sweng.erpa.model.UserProfile;
+import lombok.Data;
+import lombok.NonNull;
+
+public class MyAccountActivity extends DependencyConfigurationAgnosticActivity {
+    @Inject UserProfile userProfile;
+
+    // TODO: (Anne) add Butterknife for better readability
+    private ListView myListView;
+    private ArrayAdapter myAdapter;
+
+    private Context context;
+    private Resources resources;
+
+    private List<MyAccountButton> myAccountButtonList;
+    private List<Drawable> myDrawablesList;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (dependenciesNotReady()) return;
+        setContentView(R.layout.activity_my_account);
+        onResume();
+    }
+
+    @Override protected void onResume() {
+        super.onResume();
+
+        myListView = findViewById(R.id.myAccountLayout);
+        initializeArrays();
+
+        Stream<MyAccountButton> buttonStream = Stream.of(myAccountButtonList).filter(
+                b -> (b.getActiveForPlayer() == userProfile.getIsPlayer() ||
+                        b.getActiveForGM() == userProfile.getIsGm()));
+        Stream<Drawable> drawableStream = Stream.of(myDrawablesList);
+
+        List<Pair<MyAccountButton, Drawable>> collect = Stream.zip(buttonStream, drawableStream,
+                Pair::new).collect(Collectors.toList());
+
+        collect.add(new Pair<>(new MyAccountButton(resources.getString(R.string.profileText),
+                ProfileActivity.class, true, true),
+                context.getDrawable(R.drawable.ic_action_name)));
+
+        myAdapter = new MyAccountButtonAdapter(this, collect);
+        myListView.setAdapter(myAdapter);
+        myListView.setOnItemClickListener((l, v, position, id) -> startActivity(
+                new Intent(this, collect.get(position).getFirst().getActivityClass())));
+
+    }
+
+    private void initializeArrays() {
+        context = this.getApplicationContext();
+        resources = context.getResources();
+        myAccountButtonList = Arrays.asList(
+                new MyAccountButton(resources.getString(R.string.pendingRequestText),
+                        PendingRequestActivity.class, true, false),
+                new MyAccountButton(resources.getString(R.string.confirmedGamesText),
+                        ConfirmedGamesActivity.class, true, false),
+                new MyAccountButton(resources.getString(R.string.pastGamesText),
+                        PastGamesActivity.class, true, false),
+                new MyAccountButton(resources.getString(R.string.hostedGamesText),
+                        HostedGamesActivity.class, false, true),
+                new MyAccountButton(resources.getString(R.string.pastHostedGamesText),
+                        PastHostedGamesActivity.class, false, true)
+        );
+        myDrawablesList = Stream.rangeClosed(1, 8).map(i -> context.getDrawable(R.drawable.ic_action_name)).collect(
```

Comment:
Why do you range from 1 to 8 and not 1 to 6 (because of the D6 images) or 1 to `myAccountButtonList.size()`?

Reply:
sure